### PR TITLE
move stuff from _details to _internal for storage-common package

### PR DESCRIPTION
**Pull request by Jinming-Hu**
closes https://github.com/Azure/azure-sdk-for-cpp/issues/1869

# Pull Request Checklist

Please leverage this checklist as a reminder to address commonly occurring feedback when submitting a pull request to make sure your PR can be reviewed quickly:

See the detailed list in the [contributing guide](https://github.com/Azure/azure-sdk-for-cpp/blob/master/CONTRIBUTING.md#pull-requests).

- [x] [C++ Guidelines](https://azure.github.io/azure-sdk/cpp_introduction.html)
- [x] Doxygen docs
- [x] Unit tests
- [x] No unwanted commits/changes
- [x] Descriptive title/description
  - [x] PR is single purpose
  - [x] Related issue listed
- [x] Comments in source
- [x] No typos
- [x] Update changelog
- [x] Not work-in-progress
- [x] External references or docs updated
- [x] Self review of PR done
- [x] Any breaking changes?

### CODE REVIEW

**Review by katmsft — APPROVED** [review]

File: sdk/storage/azure-storage-common/inc/azure/storage/common/storage_common.hpp [hunk 1]
```diff
@@ -50,10 +50,10 @@ namespace Azure { namespace Storage {
     HashAlgorithm Algorithm = HashAlgorithm::Md5;
   };
 
-  namespace _detail {
+  namespace _internal {
```
Comment (ahsonkhan) [inline sdk/storage/azure-storage-common/inc/azure/storage/common/storage_common.hpp:53]: Is there anything in storage common that is private to the package and belongs in `_detail` or not being used at all, that we should remove?

Comment (Jinming-Hu) [reply sdk/storage/azure-storage-common/inc/azure/storage/common/storage_common.hpp:53]: I don't think there's anything in common package is private to this package.

**Pull request merged**